FirebaseStorage/Sources/Internal/StorageInternalTask.swift

```diff
@@ -27,35 +27,35 @@ class StorageInternalTask: StorageTask {
 
   @discardableResult
   init(reference: StorageReference,
-       fetcherService: GTMSessionFetcherService,
        queue: DispatchQueue,
        request: URLRequest? = nil,
        httpMethod: String,
        fetcherComment: String,
        completion: ((_: Data?, _: Error?) -> Void)?) {
-    super.init(reference: reference, service: fetcherService, queue: queue)
+    super.init(reference: reference, queue: queue)
 
     // Prepare a task and begins execution.
     dispatchQueue.async { [self] in
       self.state = .queueing
-      var request = request ?? self.baseRequest
-      request.httpMethod = httpMethod
-      request.timeoutInterval = self.reference.storage.maxOperationRetryTime
+      Task {
+        let fetcherService = await reference.storage.fetcherService
+          .fetcherService(reference.storage)
```

Comment:
optional/not urgent: This line makes me consider if `Storage` should even have a `fetcherService` property since it passes right through. I couldn't think of an obvious option, but a few to consider. Maybe an async var on one of the `StorageTask` class (or subclass) since they have has access to a `StorageReference`. It also may be worth making `StorageFetcherService` accessible via a singleton to prevent two classes from doing `StorageFetcherService()` and mirroring instance dictionaries.

Reply:
Interesting thought. I was trying to keep the behavior of a `fetcherService` per FirebaseApp, but not sure that's necessary.

Do you see a problem with multiple different instances of `StorageFetcherService`?

Reply:
> I was trying to keep the behavior of a fetcherService per FirebaseApp, but not sure that's necessary.

That SGTM. I was suggesting something like:

```suggestion
        let fetcherService = await reference.fetcherService
```
where in `StorageReference.swift`:
```swift
extension StorageReference {
   var fetcherService: GTMSessionFetcherService {
        get async {
            StorageFetcherService.shared.fetcherService(self.storage)
        }
    }
}
```
or maybe it fits better in `StorageTask.swift`:
```swift
extension StorageTask {
    var fetcherService: GTMSessionFetcherService {
        get async {
            StorageFetcherService.shared.fetcherService(self.reference.storage)
        }
    }
}
```

> Do you see a problem with multiple different instances of StorageFetcherService?

In practice, probably not, but IIUC multiple instances of an actor do not share synchronization so the static instance cache could have data races.

Reply:
Oops, I left the Sources out of my last `git add`. Addressed in 6f2229dca

I'll do some more research, but we made need separate fetchers if there some of the FirebaseApps are running on the emulator and some on projects.

The static instance cache race is a possibility though ...